samza-api/src/main/java/org/apache/samza/metrics/SamzaHistogram.java

```diff
@@ -60,6 +60,11 @@ public void updateGaugeValues(double percentile) {
     gauges.get(percentile).set(values.getValue(percentile / 100));
   }
 
+  //for debug
+  public Map<Double, Gauge<Double>> getGauges() {
+    return gauges;
+  }
```

Comment:
Is this only for testing?
In that case you can make it package-private and use @visiblefortesting flag.

Reply:
I will modify it and use a MetricsVisitor to get access to the Histogram instead of adding a getter method.

Reply:
I deleted this method and used a MetricsVisitor instead.